### PR TITLE
[extractor/rheinmaintv] Add extractor

### PR DESCRIPTION
**Pull request by bashonly**
Supersedes #5840

PR commit 08d65a66225b1b6cd1c2048c52bf16f8c4b1d0a5 reapplies a change made in yt-dlp:master commit a5387729696a5b33f53f60ef06f48e45663b12dd that was somehow reverted in 69bec6730ec9d724bcedeab199d9d684d61423ba - without this fix ismv+isma formats will not merge into mp4 container

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [ ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence): **almost all code was authored by [barthelmannk](https://github.com/barthelmannk)** in #5840

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
📺🛠️🐛

This pull request adds a new extractor for Rhein-Main TV videos, by creating the `RheinMainTVIE` class in the `rheinmaintv.py` module and importing it in the `_extractors.py` module. It also fixes a bug in the `get_compatible_ext` function in the `utils/_utils.py` module, by sanitizing the codec names before checking their compatibility with MP4 containers.

> _`RheinMainTVIE` added_
> _Extract videos from German channel_
> _Winter nights are bright_

### Walkthrough
*  Implement video extraction for Rhein-Main TV ([link](https://github.com/yt-dlp/yt-dlp/pull/7311/files?diff=unified&w=0#diff-b38dcea25145512ac7f8bb0e7bdb073f7f21cea288bcfcd3f190e522a3d56128R1-R97),[link](https://github.com/yt-dlp/yt-dlp/pull/7311/files?diff=unified&w=0#diff-780b22dc7eb280f5a7b2bbf79aff17826de88ddcbf2fc1116ba19901827aa4e3R1615))
   * Add `rheinmaintv.py` module that defines `RheinMainTVIE` class ([link](https://github.com/yt-dlp/yt-dlp/pull/7311/files?diff=unified&w=0#diff-b38dcea25145512ac7f8bb0e7bdb073f7f21cea288bcfcd3f190e522a3d56128R1-R97))
   * Override `_real_extract` method to parse video information and formats from webpage ([link](https://github.com/yt-dlp/yt-dlp/pull/7311/files?diff=unified&w=0#diff-b38dcea25145512ac7f8bb0e7bdb073f7f21cea288bcfcd3f190e522a3d56128R1-R97))
   * Import `RheinMainTVIE` class in `_extractors.py` and add valid URLs ([link](https://github.com/yt-dlp/yt-dlp/pull/7311/files?diff=unified&w=0#diff-780b22dc7eb280f5a7b2bbf79aff17826de88ddcbf2fc1116ba19901827aa4e3R1615))
   * Add tests for `RheinMainTVIE` in `rheinmaintv.py` ([link](https://github.com/yt-dlp/yt-dlp/pull/7311/files?diff=unified&w=0#diff-b38dcea25145512ac7f8bb0e7bdb073f7f21cea288bcfcd3f190e522a3d56128R1-R97))
* Fix bug with uppercase codecs in MP4 containers ([link](https://github.com/yt-dlp/yt-dlp/pull/7311/files?diff=unified&w=0#diff-feda8f56946dc048e754111850baaef0eec4b9f8bbc2d3f04b1a785626ea5c0eL3505-R3506))
   * Modify `sanitize_codec` function in `utils/_utils.py` to convert codecs to lowercase ([link](https://github.com/yt-dlp/yt-dlp/pull/7311/files?diff=unified&w=0#diff-feda8f56946dc048e754111850baaef0eec4b9f8bbc2d3f04b1a785626ea5c0eL3505-R3506))
   * Avoid incorrect detection of incompatible formats such as ISMV ([link](https://github.com/yt-dlp/yt-dlp/pull/7311/files?diff=unified&w=0#diff-feda8f56946dc048e754111850baaef0eec4b9f8bbc2d3f04b1a785626ea5c0eL3505-R3506))

</details>

### CODE REVIEW

File: yt_dlp/extractor/rheinmaintv.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            'alt_title': img.get('alt'),
+            'description': json_ld.get('description') or self._og_search_description(webpage),
+            'formats': formats,
+            'subtitles': subtitles,
+            'thumbnails': [{'url': img['src']}] if 'src' in img else json_ld.get('thumbnails'),
+            **traverse_obj(json_ld, {
+                'timestamp': 'timestamp',
+                'duration': 'duration',
+                'view_count': 'view_count',
+            }),
```
Comment (pukkandan) [inline yt_dlp/extractor/rheinmaintv.py:99]: Why not merge_dict with the whole thing?

File: yt_dlp/extractor/rheinmaintv.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        video_id = mobj.group('video_id').replace('/', '-')
+        webpage = self._download_webpage(url, video_id)
+
+        source, img = self._search_regex(r'(?s)(?P<source><source[^>]*>)(?P<img><img[^>]*>)',
+                                         webpage, 'video', group=('source', 'img'))
+        source = extract_attributes(source)
+        img = extract_attributes(img)
+
+        # Work around the method self._json_ld (called by self._search_json_ld), which
+        # extracts the useless 'contentUrl' (as 'url') instead of the essential 'embedUrl'.
```
Comment (pukkandan) [inline yt_dlp/extractor/rheinmaintv.py:72]:
```suggestion
```
Is a known pattern

File: yt_dlp/utils/_utils.py [hunk 3]
```diff
@@ -3502,7 +3502,8 @@ def get_compatible_ext(*, vcodecs, acodecs, vexts, aexts, preferences=None):
         },
     }
 
-    sanitize_codec = functools.partial(try_get, getter=lambda x: x[0].split('.')[0].replace('0', ''))
+    sanitize_codec = functools.partial(
+        try_get, getter=lambda x: x[0].split('.')[0].replace('0', '').lower())
```
Comment (pukkandan) [inline yt_dlp/utils/_utils.py:3506]: I'll merge this separately

**Review by pukkandan — APPROVED** [review]

File: yt_dlp/extractor/rheinmaintv.py (hunk 1 above)
Comment (bashonly) [reply yt_dlp/extractor/rheinmaintv.py:97]: I think the original PR author wanted to mix and match, they wanted to give priority to JSON LD for certain fields but also wanted non-JSON LD fallbacks for those (`description`, `title`)

But yeah ig merging the whole thing with JSON LD as secondary dict couldn't hurt

**Pull request merged**